src/hotspot/share/runtime/os.hpp

```diff
@@ -224,6 +224,13 @@ class os: AllStatic {
   static char*      local_time_string(char *buf, size_t buflen);
   static struct tm* localtime_pd     (const time_t* clock, struct tm*  res);
   static struct tm* gmtime_pd        (const time_t* clock, struct tm*  res);
+
+  // Fill in buffer with an ISO-8601 string corresponding to the given javaTimeMillis value
+  // E.g., YYYY-MM-DDThh:mm:ss.mmm+zzzz.
+  // Returns buffer, or NULL if it failed.
+  static char* iso8601_time(jlong milliseconds_since_19700101, char buffer[29],
```

Comment:
You are supposed to get a compiler warning on this char buffer[29]. ISO C/C++ can't pass in array type as a function argument.  As a result, the type buffer automatically casts to char* here. 

Further, the function signature is different from its implementation, which use char* buffer instead. 
I see that you do enforce length checking in implementation.

Reply:
Oh, good catch, this is a stray edit I thought I reverted. See below.

> ISO C/C++ can't pass in array type as a function argument. As a result, the type buffer automatically casts to char* here.

No, it can, that's totally fine. It gets automatically translated to pointer type, but its a valid expression nonetheless. 

It is usually done to indicate that a function expects an output array of a particular length instead of letting the caller guess the right length - and letting him to deal with truncation if he guesses wrong.

I originally planned to define the function as 
```
iso8601_time(jlong milliseconds_since_19700101, char buffer[29]);
```
to indicate that the only output value length which makes sense is 29, but then I refrained because I wanted to keep the patch simple.

Reply:
C/C++  allow you to write code like that, but they won't check the length of argument <=29 for you. 
I think you sanity check inside of iso8601_time() is good enough.